src/runner.ts

```diff
@@ -0,0 +1,86 @@
+import * as core from "@actions/core";
+
+export type CoberturaCoverageNode = {
+    lineRate: number;
+    linesCovered: number;
+    linesValid: number;
+    packages: Map<string, CoberturaPackageNode>;
+}
+
+export type CoberturaPackageNode = {
+    name: string;
+    lineRate: number;
+    classes: Map<string, CoberturaClassNode>;
+}
+
+export type CoberturaClassNode = {
+    classId: string; // Use "name + filename" to identify the class
+    fileName: string;
+    name: string;
+    lineRate: number;
+    coveredLines: number;
+    lines: CoberturaLineNode[];
+}
+
+type CoberturaLineNode = {
+    lineNumber: number;
+    lineHash: string;
+    hits: number;
+}
+
+export class CoverageParserRunner {
+    // TODO: Implement converting Parasoft coverage XML report to cobertura report
+    // TODO: Implement calculating coverage data from the converted report
+    async customizeJobRunSummary(coverageNode: CoberturaCoverageNode) {
+        try {
+            const markdown = this.customizeMarkdownContent(coverageNode.packages);
+            const totalCoverage = this.formatCoverage(coverageNode.linesCovered, coverageNode.linesValid, coverageNode.lineRate);
+
+            return await core.summary
+                .addRaw("<table><tbody><tr><th>Coverage&emsp;(covered/total - percentage)</th></tr>"
+                    + "<tr><td><b>Total coverage&emsp;(" + totalCoverage + ")</b></td></tr>"
+                    + markdown + "</tbody></table>")
+                .write();
+        } catch (error) {
+            console.error("An error occurred while customizing the job run summary:", error);
+        }
+    }
+
+    private customizeMarkdownContent(packagesNode: Map<string, CoberturaPackageNode>) {
+        return Array.from(packagesNode.entries()).map(([packageName, packageNode]) => {
```

Comment:
`null` and empty check for `packagesNode` should be added at the beginning of this method.